packages/components/src/spectrum/CheckboxGroup.tsx

```diff
@@ -0,0 +1,34 @@
+import {
+  CheckboxGroup as SpectrumCheckboxGroup,
+  SpectrumCheckboxGroupProps,
+} from '@adobe/react-spectrum';
+import { isElementOfType } from '@deephaven/react-hooks';
+import React, { useMemo } from 'react';
+import Checkbox from '../Checkbox';
+
+/**
+ * Augmented version of the Spectrum CheckboxGroup component that supports
+ * primitive item children.
+ */
+export function CheckboxGroup({
+  children,
+  ...props
+}: SpectrumCheckboxGroupProps): JSX.Element {
+  const wrappedChildren = useMemo(
+    () =>
+      React.Children.map(children, child => {
+        if (isElementOfType(child, Checkbox)) {
+          return child;
+        }
+        return <Checkbox checked={false}>{child}</Checkbox>;
```

Comment:
I think this needs to be the Spectrum CheckBox

Reply:
Also, I'd still highly recommend trying some examples in the Styleguide. As-is, I don't think it actually accepts primitive children.